subworkflows/local/prepare_genome.nf

```diff
@@ -27,71 +27,45 @@ workflow PREPARE_GENOME {
+    (ch_bwa2, ch_bwa_version) = BWAMEM2_INDEX(fasta)
 
-    ch_dict = Channel.empty()
-    if (!(params.dict) && !('annotate' in step) && !('controlfreec' in step)) {
-        GATK4_CREATESEQUENCEDICTIONARY(fasta)
-        ch_dict = GATK4_CREATESEQUENCEDICTIONARY.out.dict
-        ch_versions = ch_versions.mix(GATK4_CREATESEQUENCEDICTIONARY.out.versions)
-    }
+    ch_bwa = ch_bwa1.mix(ch_bwa2)
+    ch_versions = ch_versions.mix(ch_bwa_version)
 
-    ch_fasta_fai = Channel.empty()
-    if (fasta_fai) ch_fasta_fai = fasta_fai
-    if (!(params.fasta_fai) && !('annotate' in step)) {
-        SAMTOOLS_FAIDX(fasta.map{ it -> [[id:it[0].getName()], it]})
-        ch_fasta_fai = SAMTOOLS_FAIDX.out.fai.map{ meta, fai -> [fai] }
-        ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)
-    }
+    (ch_dict, ch_dict_version) = GATK4_CREATESEQUENCEDICTIONARY(fasta)
 
-    ch_dbsnp_tbi = Channel.empty()
-    if (!(params.dbsnp_tbi) && params.dbsnp && ('mapping' in step || 'prepare_recalibration' in step || tools.contains('controlfreec') || tools.contains('haplotypecaller') || tools.contains('mutect2') || tools.contains('tnscope'))) {
-        TABIX_DBSNP(dbsnp.map{ it -> [[id:it[0].baseName], it] })
-        ch_dbsnp_tbi = TABIX_DBSNP.out.tbi.map{ meta, tbi -> [tbi] }
-        ch_versions = ch_versions.mix(TABIX_DBSNP.out.versions)
-    }
+    ch_versions = ch_versions.mix(ch_dict_version)
 
-    ch_germline_resource_tbi = Channel.empty()
-    if (!(params.germline_resource_tbi) && params.germline_resource && tools.contains('mutect2')) {
-        TABIX_GERMLINE_RESOURCE(germline_resource.map{ it -> [[id:it[0].baseName], it] })
-        ch_germline_resource_tbi = TABIX_GERMLINE_RESOURCE.out.tbi.map{ meta, tbi -> [tbi] }
-        ch_versions = ch_versions.mix(TABIX_GERMLINE_RESOURCE.out.versions)
+    if (fasta_fai) ch_fasta_fai = fasta_fai
+    else {
+        (ch_fasta_fai, ch_fasta_fai_version) = SAMTOOLS_FAIDX(fasta.map{ it -> [[id:it[0].getName()], it]})
+        ch_fasta_fai = ch_fasta_fai.map{ meta, fai -> [fai] }
+        ch_versions = ch_versions.mix(ch_fasta_fai_version)
     }
 
-    ch_known_indels_tbi = Channel.empty()
-    if (!(params.known_indels_tbi) && params.known_indels && ('mapping' in step || 'prepare_recalibration' in step)) {
-        TABIX_KNOWN_INDELS(known_indels.map{ it -> [[id:it[0].baseName], it] })
-        ch_known_indels_tbi = TABIX_KNOWN_INDELS.out.tbi.map{ meta, tbi -> [tbi] }
-        ch_versions = ch_versions.mix(TABIX_KNOWN_INDELS.out.versions)
-    }
+    (ch_dbsnp_tbi,             ch_dbsnp_tbi_version)             = TABIX_DBSNP(dbsnp.map{ it -> [[id:it[0].baseName], it] })
+    (ch_germline_resource_tbi, ch_germline_resource_tbi_version) = TABIX_GERMLINE_RESOURCE(germline_resource.map{ it -> [[id:it[0].baseName], it] })
+    (ch_known_indels_tbi,      ch_known_indels_tbi_version)      = TABIX_KNOWN_INDELS(known_indels.map{ it -> [[id:it[0].baseName], it] })
```

Comment:
this is awesome, didn't know it works <3

Reply:
I saw @mahesh-panchal draft PR somewhere, so I knew I wanted to try it